packages/core/wallet/src/initialize.ts

```diff
@@ -0,0 +1,41 @@
+import type {
+    Wallet,
+    WalletEventsWindow,
+    WalletInitializeEvent as WalletInitializeEventInterface,
+    WalletInitializeEventCallback,
+} from '@wallet-standard/base';
+
+declare const window: WalletEventsWindow;
+
+export function initializeWallet(wallet: Wallet): void {
+    const callback: WalletInitializeEventCallback = ({ register }) => register(wallet);
+    window.dispatchEvent(new WalletInitializeEvent(callback));
+    window.addEventListener('wallet-standard-app-initialize', ({ detail: api }) => callback(api));
```

Comment:
I wonder if different names would lend themselves more easily to folks understanding, intuitively, what's going on here.

The event that you optimistically fire is to register your wallet to anyone who's listening. Maybe `RegisterWalletEvent` (ie. `wallet-standard:register-wallet`)?

The event that you subscribe is to learn of future things you might like to register your wallet with. Maybe `WalletRegistrarRequestForWalletEvent` (ie. `wallet-standard:registrar-request-for-wallet`)?

I guess what I'm getting stuck on is the event names being so high-level (ie. ‘initialize’).

Reply:
Yeah, `initialize` is very broad. The thinking is that each wallet and app will "initialize" or "set up" once per. Does it actually care about what's happening inside there? It's supposed to be abstracted away. `register` makes it seem like it's registering _now_ but really what it's doing is setting up scaffolding to register later. Not a sticking point for me, though the idea for wallets and apps is "don't think about it too hard, just call this once".